rbac/management/group/definer.py

```diff
@@ -0,0 +1,88 @@
+#
+# Copyright 2019 Red Hat, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+
+"""Handler for system defined group."""
+import logging
+
+from django.core.exceptions import ValidationError
+from management.group.model import Group
+from management.policy.model import Policy
+from management.role.model import Role
+from tenant_schemas.utils import tenant_context
+
+logger = logging.getLogger(__name__)  # pylint: disable=invalid-name
+
+
+def seed_group(tenant):
+    """For a tenant create or update default group."""
+    with tenant_context(tenant):
+        name = 'Platform group'
+        group_description = 'Default group that contains default permissions for new users.'
+
+        group, group_created = Group.objects.get_or_create(
+            platform_default=True,
+            defaults={
+                'description': group_description,
+                'name': name,
+                'system': True
+            }
+        )
+
+        if group.system:
+            platform_roles = Role.objects.filter(platform_default=True).values_list('uuid', flat=True)
+            add_roles(group, platform_roles, replace=True)
+            logger.info('Finished seeding default group %s for tenant %s.', name, tenant.schema_name)
+        else:
+            logger.info('Default group %s is managed by tenant %s.', name, tenant.schema_name)
+    return tenant
+
+
+def add_roles(group, roles, replace=False):
+    """Process list of roles and add them to the group."""
+    system_policy_name = 'System Policy for Group {}'.format(group.uuid)
+    system_policy, system_policy_created = Policy.objects.get_or_create(system=True,
+                                                                        group=group,
+                                                                        name=system_policy_name)
+    if system_policy_created:
+        logger.info('Created new system policy for tenant.')
+    else:
+        if replace:
+            system_policy.roles.clear()
```

Comment:
So this will remove all roles, but only for the platform group, correct? If the tenant has another group with custom roles that have been added, it would not be affected by this, since we're only ever replacing roles on the system policy for the platform default group.

Reply:
Yes, so the default for "replace" flag is false. For other groups other than the default group, we would just use default value, which would not. We would only set this to true when adding roles to default group.

I was thinking making the function for any one who might be interested in replacing the roles.

Do you want me to add another condition such as group.system is true to enforce this could only apply to platform group?

Reply:
I don't think that's necessary, and might actually couple things too much to that implementation. I just more wanted to confirm there wouldn't be any unintended side-effects. 😄